rasa_nlu/extractors/crf_entity_extractor.py

```diff
@@ -141,11 +141,29 @@ def extract_entities(self, message):
         if self.ent_tagger is not None:
             text_data = self._from_text_to_crf(message)
             features = self._sentence_to_features(text_data)
-            ents = self.ent_tagger.predict_single(features)
+            ents = self.ent_tagger.predict_marginals_single(features)
             return self._from_crf_to_json(message, ents)
         else:
             return []
 
+    def most_likely_entity(self, idx, entities):
+        entity_probs = entities[idx]
+        if entity_probs:
+            label = max(entity_probs,
+                         key=lambda key: entity_probs[key])
+            if self.BILOU_flag:
+                # if we are using bilou flags, we will combine the prob
+                # of the B, I, L and U tags for an entity (so if we have a
+                # score of 60% for `B-address` and 40% and 30%
+                # for `I-address`, we will return 70%)
+                return label, sum([v
+                                   for k, v in entity_probs.items()
+                                   if k[2:] == label[2:]])
```

Comment:
I'm a bit confused by your comment, wouldn't you in that case have the label `B-address` but use the probability 70% which corresponds to adding the I-address labels?

Reply:
Yes, you'd use the `B-address` label but use p=0.7 as probability instead of 0.4 - I mean we can change this, but the idea is that the model thinks it is the `address` entity no matter what BILOU tag it assigns it (e.g. it might be correct in predicting the address type but assigning a B where an U might be more appropriate and so on)

Reply:
ah ok, that's fine then